### PR TITLE
refactor(icon): Update CSS styles to match React implementation

### PR DESCRIPTION
**Pull request by jamesfan**
This PR updates the Icon CSS styles so they match the React component. 

|                  | Visual | Code |
|------------------|--------|------|
| Breaking Changes | No    | No   |

### Squash and Merge Message Proposal
refactor(icon): Update CSS styles to match React implementation (#159)

### CODE REVIEW

**Comment by jamesfan** [comment]
Still a WIP (working on `color` and `hover` support for System Icons), but wanted to get this PR out there for preliminary review.

**Comment by jamesfan** [comment]
Here's a list of the React Icon functionality which has been ported to the CSS Icon (everything is now supported with the exception of the `data-color` shorthand for the System Icon):

All Icons
- [x] `data-size`

Accent Icons
- [x] `data-transparent`

System Icons
- [x] `data-circle`
- [x] `data-circle-background`
- [x] `data-hover-fill-color`
- [x] `data-hover-accent-color`
- [x] `data-hover-background-color`
- [x] `data-hover-color` (shorthand for `data-hover-fill-color` and `data-hover-accent-color`)
- [ ] `data-color` (shorthand for `data-fill-color` and `data-accent-color`) _In progress_

@anicholls I'm working on adding support for `data-color` for System Icon. In the meantime, let me know if you see any issues or if you have any suggestions regarding code style.

File: modules/icon/css/README.md [hunk 1]
```diff
@@ -92,36 +162,72 @@ the 500 weight (e.g. `blueberry500`).
...
+
+### System Icon Circles
 
 ```html
-<i className="wdc-icon" data-icon="rocket" data-category="applet" data-color="cinnamon" />
+<i
+  className="wdc-icon"
+  data-icon="activityStream"
+  data-category="system"
+  data-circle="true"
```
Comment (anicholls) [inline modules/icon/css/README.md]: Can we just assume this if `data-circle-background` is defined and remove the default?

File: modules/icon/css/stories.tsx [hunk 2]
```diff
@@ -2,45 +2,111 @@ import React, {Component} from 'react';
...
-  }
-}
-
-storiesOf('CSS/Icon', module)
-  .addDecorator(withReadme(README))
-  .add('All', () => (
-    <div className="story">
-      <IconDemo />
+        <br />
+        <i className="wdc-icon" data-icon="activityStream" data-category="system" data-size="48" />
```
Comment (anicholls) [inline modules/icon/css/stories.tsx]: Can we add an example with hover colors?

File: modules/icon/css/README.md (hunk 1 above)
Comment (jamesfan) [reply modules/icon/css/README.md]: Just to clarify: `data-circle="true"` on its own will render the circle with its default background color, but you can also do `data-circle-background="COLOR"` on its own (**without** `data-circle="true"`) to render the circle in a non-default color. Is that correct?

File: modules/icon/css/stories.tsx (hunk 2 above)
Comment (jamesfan) [reply modules/icon/css/stories.tsx]: Sounds good. I have a handful of stories I was using to test the hover colors anyway -- I'll use those.

File: modules/icon/css/README.md (hunk 1 above)
Comment (anicholls) [reply modules/icon/css/README.md]: Ah - should have looked closer at the logic. Either way, I would consolidate. You could just use `data-circle-background="true"` to get the default, and a canvas color, or custom hex to get the non-default.

Comment (jamesfan) [reply modules/icon/css/README.md]: Got it, will do.

File: modules/icon/css/stories.tsx (hunk 2 above)
Comment (jamesfan) [reply modules/icon/css/stories.tsx]: Quick question: should I add stories with non-default hover colors to the React System Icon as well? Or does that belong in a separate PR?

Comment (anicholls) [reply modules/icon/css/stories.tsx]: You could probably do it here - that's fine 👍

**Comment by jamesfan** [comment]
@anicholls the following items should be addressed:

- [x] Remove `data-circle` and update `data-circle-background` to accept `true` for the default background color
- [x] Add examples with hover colors (CSS and React)

I also improved the formatting/structure of the `README`. It was getting a bit hard to read with all of the `data` attributes -- I followed a similar format to the `README` for the React Icon.

**Pull request merged**